.github/workflows/main.yml

```diff
@@ -113,6 +113,28 @@ jobs:
             .ci/check-format.sh
       shell: bash
   
+  static-analysis:
+    needs: [detect-code-related-file-changes]
+    if: needs.detect-code-related-file-changes.outputs.has_code_related_changes == 'true'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    # LLVM static analysis
+    - name: set up scan-build  
+      run: |
+            sudo apt-get update -q -y
+            sudo apt-get install -q -y clang clang-tools libsdl2-dev libsdl2-mixer-dev
+      shell: bash
+    - name: run scan-build
+      run: scan-build -v -o ~/scan-build --use-cc=clang --force-analyze-debug-code --show-description -analyzer-config stable-report-filename=true -enable-checker valist,nullability make ENABLE_JIT=1
```

Comment:
We can prepare a shell script to launch `scan-build` in `.ci` directory and later execute when needded.